src/Microsoft.Diagnostics.EventFlow.Core/Implementations/DiagnosticPipeline.cs

```diff
@@ -13,18 +13,20 @@
 using Validation;
 
 using Microsoft.Diagnostics.EventFlow.Configuration;
+using System.Diagnostics;
 
 namespace Microsoft.Diagnostics.EventFlow
 {
     public class DiagnosticPipeline : IDisposable
     {
-        private List<IDisposable> pipelineDisposables;
+        private List<IDisposable> pipelineLinkDisposables;
         private List<IDisposable> inputSubscriptions;
         private bool disposed;
         private bool disposeDependencies;
         private CancellationTokenSource cancellationTokenSource;
         private IDataflowBlock pipelineHead;
         private DiagnosticPipelineConfiguration pipelineConfiguration;
+        private volatile int eventsInProgress;
```

Comment:
I feel it's hard to maintain this variable across blocks. Ideally it's supported by DataFlow natively.
Is it possible to have such a variable for each block, and dispose from the head to the tail?